js/bitmex.js

```diff
@@ -2666,6 +2697,46 @@ module.exports = class bitmex extends Exchange {
         return await this.privatePostPositionIsolate (this.extend (request, params));
     }
 
+    async fetchDepositAddress (code, params = {}) {
+        /**
+         * @method
+         * @name bitmex#fetchDepositAddress
+         * @description fetch the deposit address for a currency associated with this account
+         * @see https://www.bitmex.com/api/explorer/#!/User/User_getDepositAddress
+         * @param {string} code unified currency code
+         * @param {object} params extra parameters specific to the bitmex api endpoint
+         * @param {string} params.network deposit chain, can view all chains via this.publicGetWalletAssets, default is eth, unless the currency has a default chain within this.options['networks']
+         * @returns {object} an [address structure]{@link https://docs.ccxt.com/en/latest/manual.html#address-structure}
+         */
+        await this.loadMarkets ();
+        const networkCode = this.safeStringUpper (params, 'network');
+        if (networkCode === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchDepositAddress requires params["network"]');
+        }
+        const networks = this.safeValue (this.options, 'networks', {});
+        const networkId = this.safeStringLower (networks, networkCode, networkCode);
+        const currency = this.currency (code);
+        let currencyId = currency['id'];
+        const idLength = currencyId.length;
+        currencyId = currencyId.slice (0, idLength - 1) + currencyId.slice (idLength - 1, idLength).toLowerCase ();  // make the last letter lowercase
```

Comment:
@ttodua tagging you so we don't forget after your merge, we can remove this logic because the id will be correct. But don't worry we can do this in a separate PR

Reply:
The ids were usually something like `ETH`, but the api endpoint usually requires something like `ETh` for some reason, and it was only this endpoint

Reply:
@carlosmiei I think my bitmex PR includes all fixes for bitmex, related to currencies & networks & markets.

Reply:
> @carlosmiei I think my bitmex PR includes all fixes for bitmex, related to currencies & networks & markets.

This endpoint specifically requires all uppercase and the last letter to be lowercase, but other ones don't, so I don't see how you could have written a fix for this issue